mongo/mongo.go

```diff
@@ -435,26 +441,38 @@ func EnsureServer(args EnsureServerParams) (Version, error) {
 
 func ensureServer(args EnsureServerParams, mongoKernelTweaks map[string]string) (Version, error) {
 	var zeroVersion Version
+	var err error
 	tweakSysctlForMongo(mongoKernelTweaks)
 	logger.Infof(
 		"Ensuring mongo server is running; data directory %s; port %d",
 		args.DataDir, args.StatePort,
 	)
 
+	if featureflag.Enabled(feature.MongoDbSnap) {
+		if args.DataDir != snap.DefaultDataPath {
+			logger.Warningf("overwriting args.dataDir (set to %v) to %v", args.DataDir, snap.DefaultDataPath)
+			args.DataDir = snap.DefaultDataPath
+		}
+	}
+
 	dbDir := filepath.Join(args.DataDir, "db")
 	if err := os.MkdirAll(dbDir, 0700); err != nil {
 		return zeroVersion, errors.Errorf("cannot create mongo database directory: %v", err)
 	}
 
 	oplogSizeMB := args.OplogSize
 	if oplogSizeMB == 0 {
-		var err error
 		if oplogSizeMB, err = defaultOplogSize(dbDir); err != nil {
 			return zeroVersion, errors.Trace(err)
 		}
 	}
 
-	if err := installMongod(series.MustHostSeries(), args.SetNUMAControlPolicy); err != nil {
+	series := series.MustHostSeries()
+	if featureflag.Enabled(feature.MongoDbSnap) {
+		series = "snap"
```

Comment:
let's not abuse series in this way
You need to introduce a decision point here for snap vs package.
installMongod() below is very package specific in all it does.
Consider a new installMongodSnap() func to call for a snap to do snappy type things